### ISSUE TITLE
"Run current cell and go to the next one" does not always highlight next cell

### ISSUE THREAD

**Issue by remisalmon** [spyder-ide/spyder#11712]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

When clicking "Run current cell and go to the next one" with "Highlight current cell" activated, only one every other cell is highlighted. This issue was also present on spyder 4.0.0 but not on spyder 3.3.6.

### What steps reproduce the problem?

1. Create multiple cells
2. Select a cell, click "Run current cell and go to the next one" 
3. The cursor moves to the next cell but the previous cell is still highlighted
4. Click again "Run current cell and go to the next one"
5. The correct cell is highlighted

### What is the expected output? What do you see instead?

See:
![bug](https://user-images.githubusercontent.com/26257211/75800879-796d1d80-5d37-11ea-9c20-86e5671da9a2.gif)

## Versions

* Spyder version: 4.0.1
* Python version: 3.6.7
* Qt version: 5.9.6
* PyQt version: 5.9.2
* Operating System name/version: Linux 5.5.7

### Dependencies

```
atomicwrites >=1.2.0         :  1.3.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
cloudpickle >=0.5.0          :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
IPython >=4.0                :  7.12.0 (OK)
jedi =0.14.1                 :  0.14.1 (OK)
keyring                      :  None (OK)
nbconvert >=4.0              :  5.6.1 (OK)
numpydoc >=0.6.0             :  0.9.2 (OK)
pexpect >=4.4.0              :  4.8.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
psutil >=0.3                 :  5.6.7 (OK)
pygments >=2.0               :  2.5.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.7 (OK)
xdg >=0.26                   :  0.26 (OK)
zmq >=17                     :  18.1.1 (OK)
qdarkstyle >=2.7             :  2.8 (OK)
qtawesome >=0.5.7            :  0.6.1 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
rtree >=0.8.3                :  0.9.3 (OK)
sphinx >=0.6.6               :  2.4.0 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
watchdog                     :  None (OK)
cython >=0.21                :  None (NOK)
matplotlib >=2.0.0           :  3.1.3 (OK)
numpy >=1.7                  :  1.18.1 (OK)
pandas >=0.13.1              :  None (NOK)
scipy >=0.17.0               :  1.4.1 (OK)
sympy >=0.7.3                :  None (NOK)
```

**Comment by goanpeca** [comment]
I could reproduce this on 4.0.1 but I cannot reproduce this on OSX with the latest 4.x branch  so I think this has been fixed already.

**Comment by goanpeca** [comment]
@remisalmon thanks for reporting, this has already been fixed for 4.1.0 version to be released in a couple of weeks

**Comment by remisalmon** [comment]
Great, thanks for the good work on spyder!
